src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -1397,7 +1397,7 @@ public boolean allowsEntityEventCreation() {
 
     @Override
     public void refreshCache() {
-        if (this.entityType != null) {
+        if (this.entityType instanceof SpongeEntityType) {
```

Comment:
Quoting gabizou `EntityTypes.UNKNOWN should never be found or used, even when mods are present.`

IMO `this.entityType != null` is fine, it avoids a NPE.

Reply:
It does not avoid an NPE because `entityType` is never null, as said in the description. `this.entityType` is initialized through `Registry#getTranslation`, which *never* returns null and instead returns `EntityTypes.UNKNOWN`, which does not extend from `SpongeEntityType`.

If the `entityType` should not be unknown at this point, then this should be replaced with a `Preconditions#checkState` verification.

Reply:
> this.entityType is initialized through Registry#getTranslation

Oh I see, I got entity and entityType mixed up.

> If the entityType should not be unknown at this point, then this should be replaced with a Preconditions#checkState verification.

What about logging more information about the entity (like position, etc) ? Do we really have to crash the server ?

Reply:
What about logging more information about the entity (like position, etc) ? Do we really have to crash the server ?

dump the class name, it will name and shame the entity / mod doing registration incorrectly (or reveal a flaw in our system)